.eslintrc

```diff
@@ -39,13 +39,51 @@
     ],
     "@typescript-eslint/camelcase": "off",
     "@typescript-eslint/consistent-indexed-object-style": "error",
+    "@typescript-eslint/consistent-type-assertions": "error",
     "@typescript-eslint/consistent-type-definitions": ["error", "interface"],
     "@typescript-eslint/explicit-function-return-type": "off",
     "@typescript-eslint/explicit-module-boundary-types": "off",
+    "@typescript-eslint/member-ordering": [
+      "error",
+      {
+        "default": {
+          "memberTypes": ["field", "constructor", "method", "signature"],
+          "order": "alphabetically"
+        }
+      }
+    ],
     "@typescript-eslint/naming-convention": "off",
+    "@typescript-eslint/no-base-to-string": "error",
+    "@typescript-eslint/no-confusing-non-null-assertion": "error",
+    "@typescript-eslint/no-confusing-void-expression": "error",
     "@typescript-eslint/no-duplicate-imports": "error",
+    "@typescript-eslint/no-dynamic-delete": "error",
+    "@typescript-eslint/no-extraneous-class": "error",
+    "@typescript-eslint/no-implicit-any-catch": "error",
+    "@typescript-eslint/no-invalid-void-type": "error",
+    "@typescript-eslint/non-nullable-type-assertion-style": "off",
+    "@typescript-eslint/no-parameter-properties": "error",
+    "@typescript-eslint/no-require-imports": "error",
     "@typescript-eslint/no-shadow": "error",
+    "@typescript-eslint/no-unnecessary-boolean-literal-compare": "error",
+    "@typescript-eslint/no-unnecessary-condition": "error",
+    "@typescript-eslint/no-unnecessary-qualifier": "off",
+    "@typescript-eslint/no-unnecessary-type-arguments": "off",
+    "@typescript-eslint/no-unnecessary-type-constraint": "error",
     "@typescript-eslint/no-use-before-define": "error",
+    "@typescript-eslint/prefer-includes": "error",
+    "@typescript-eslint/prefer-nullish-coalescing": "error",
+    "@typescript-eslint/prefer-optional-chain": "error",
+    "@typescript-eslint/prefer-readonly": "off",
+    "@typescript-eslint/prefer-readonly-parameter-types": "off",
```

Comment:
I think it would be nice to pull this up from the eslint rule into the type space.  Especially because the eslint rule doesn't handle some edge cases (reassiging the parameter, array manipulation, etc).  Plus it's technically stricter/more accurate typing.

Reply:
I wonder if we could experiment with enforcing this only on `/models/**/*.ts`, and only using a `warn`? That would be a `PocketDerm`-only implementation. 

I think my concern is that while I understand it makes our code more documented/expressive, I don't know if adding that information would actually mean fewer bugs / confusion about functionality. Have you seen otherwise? (Or is that even a fair characterization?) I am just skeptical that the below is any meaningful difference:

```diff
export interface SelectorButtonProps {
+  readonly checked: boolean;
+  readonly children?: React.ReactNode;
+  readonly disabled?: boolean;
+  readonly icon?: React.ReactNode;
+  readonly onClick?: (
    event: React.MouseEvent<HTMLDivElement, MouseEvent> | React.KeyboardEvent,
  ) => void;
+  readonly selector?: SelectorType;
+  readonly size?: SizeType;
+  readonly type?: StyleType;
+  readonly [key: string]: unknown;
-  readonly checked: boolean;
-  readonly children?: React.ReactNode;
-  readonly disabled?: boolean;
-  readonly icon?: React.ReactNode;
-  readonly onClick?: (
    event: React.MouseEvent<HTMLDivElement, MouseEvent> | React.KeyboardEvent,
  ) => void;
-  readonly selector?: SelectorType;
-  readonly size?: SizeType;
-  readonly type?: StyleType;
-  readonly [key: string]: unknown;
}